src/app/components/reports/pages/event-summary/view/view.component.ts

```diff
@@ -0,0 +1,216 @@
+import { EventSummaryReport, IEventGroup } from "@models/EventSummaryReport";
+import { User } from "@models/User";
+import { Project } from "@models/Project";
+import { Region } from "@models/Region";
+import { Site } from "@models/Site";
+import { DateTime } from "luxon";
+import { BawSessionService } from "@baw-api/baw-session.service";
+import { eventSummaryResolvers } from "@baw-api/reports/event-report/event-summary-report.service";
+import { takeUntil } from "rxjs";
+import { API_ROOT } from "@services/config/config.tokens";
+import { BinSize, EventSummaryReportParameters } from "../EventSummaryReportParameters";
+import speciesAccumulationCurveSchema from "./speciesAccumulationCurve.schema.json";
+import speciesCompositionCurveSchema from "./speciesCompositionCurve.schema.json";
+import confidencePlotSchema from "./confidencePlot.schema.json";
+
+const projectKey = "project";
+const regionKey = "region";
+const siteKey = "site";
+const reportKey = "report";
+
+@Component({
+  selector: "baw-summary-report",
+  templateUrl: "./view.component.html",
+  styleUrls: ["./view.component.scss"],
+})
+class ViewEventReportComponent
+  extends PageComponent
+  implements OnInit {
+  public constructor(
+    private route: ActivatedRoute,
+    private session: BawSessionService,
+    @Inject(API_ROOT) private apiRoot: string
+  ) {
+    super();
+  }
+
+  public parameterDataModel: EventSummaryReportParameters;
+  public report: EventSummaryReport;
+  public user: User;
+  public project: Project;
+  public region?: Region;
+  public site?: Site;
+
+  protected speciesAccumulationCurveSchema = speciesAccumulationCurveSchema;
+  protected speciesCompositionCurveSchema = speciesCompositionCurveSchema;
+  protected confidencePlotSchema = confidencePlotSchema;
+
+  public ngOnInit(): void {
+    // we can use "as" here to provide stronger typing because the data property is a standard object type without any typing
+    const models: ResolvedModelList = retrieveResolvers(this.route.snapshot.data as IPageInfo);
+
+    this.route.queryParams
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe((parameters: Params) => {
+        this.parameterDataModel = new EventSummaryReportParameters(parameters);
+      });
+
+    if (!hasResolvedSuccessfully(models)) {
+      return;
+    }
+
+    if (models[projectKey]) {
+      this.project = models[projectKey] as Project;
+    }
+
+    if (models[regionKey]) {
+      this.region = models[regionKey] as Region;
+    }
+
+    if (models[siteKey]) {
+      this.site = models[siteKey] as Site;
+    }
+
+    if (models[reportKey]) {
+      this.report = models[reportKey] as EventSummaryReport;
+    }
+  }
+
+  protected printPage(): void {
+    window.print();
+  }
+
+  protected get eventDownloadUrl(): string {
+    const base64Filters: string = this.parameterDataModel?.toFilterString();
+    return `${this.apiRoot}/projects/1135/audio_events/download.csv/?filters=${base64Filters}`;
+  }
```

Comment:
I've had a think about this and:

I was initially under the assumption that the "download csv" links (which these methods were used for) downloaded the events csv from the server (which is an existing endpoint). However, as discussed the "download csv" buttons should download element data
e.g. Event table as a csv, the graph data as csv, etc...

From my knowledge, we can't download parts of a server side model as csv (eg. charts), so we'd probably have to create a new route for downloading each elements csv data (eg. `/reports/event-summary/coverage-data/csv`).  
_Note: we could probably do the events table download with the existing events route as it supports csv responses_

Since we already have all the data client side, it might be easier to convert graph data to csv client side instead of requesting to download the data as a server created csv.

If we're going to create these csv files client side, I'm assuming we wouldn't use a service, but invent some new helper methods (eg. `vegaLiteToCsv`).

I'd probably still be able to use the `/audio_recordings/:audio_recording_id/audio_events/download.csv` route for the events table though